assets/js/base/hooks/use-store-cart-item.js

```diff
@@ -22,32 +24,43 @@ import { useStoreCart } from './use-store-cart';
  */
 export const useStoreCartItem = ( cartItemKey ) => {
 	const { cartItems, cartIsLoading } = useStoreCart();
-	const cartItem = cartItems.filter( ( item ) => item.key === cartItemKey );
+	const cartItem = cartItems.find( ( item ) => item.key === cartItemKey );
```

Comment:
we can avoid this by simply passing the cartItems to the cart, this should allow us to reload the data again (since we initially loaded it anyway), however, I'm not fully sure this is the best approach (do we want the hook to get the full item, or simply they key, I think it should be either not passing anything or passing everything).

```suggestion
export const useStoreCartItem = ( cartItem ) => {
```

Reply:
> we can avoid this by simply passing the cartItems to the cart, this should allow us to reload the data again (since we initially loaded it anyway)

This sounds a bit confusing initially, but I think what you're suggesting here is to avoid an unnecessary request triggered by implementing `useStoreCart`, you have `useStoreCartItem` receive the `cartItem` directly and then it doesn't need to implement `useStoreCart` (as evidenced from your example).

However, the primary benefit of this hook is to derive the cart item from a given cart item key and return it (so component code doesn't have to have that logic).

`useStoreCart` can be modified so that it reads the context from something like `useCheckoutContext` (as per [this pull](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/1780) being evaluated) and that determines whether it returns a default preview cart object or not.  So in my opinion, `useStoreCart` should be making the determination on whether requests are made.

To be clear, this isn't a blocking discussion for this pull.